test/integration/acceptance/grant_permission_test.cpp

```diff
@@ -0,0 +1,884 @@
+                                    kAccount2,
+                                    kEmptyPermissionName),
+              checkStatelessInvalid)
+      .done();
+}
+
+/**
+ * C258 Grant permission name, which does not exist
+ * @given an account with rights to grant rights to other accounts
+ * @when the account grants rights to an existing account
+ * AND a granted permission is non-existent
+ * @then this transaction is stateless invalid
+ * AND it is not written in the block
+ */
+TEST_F(GrantPermissionTest, GrantNonexistentPermission) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeAccountWithPerms(kAccount1,
+                                   kAccount1Keypair,
+                                   kCanGrantAll,
+                                   kRole1))
+      .skipProposal()
+      .skipBlock()
+      .sendTx(makeAccountWithPerms(kAccount2,
+                                   kAccount2Keypair,
+                                   kCanReceive,
+                                   kRole2))
+      .skipProposal()
+      .skipBlock()
+      .sendTx(accountGrantToAccount(kAccount1,
+                                    kAccount1Keypair,
+                                    kAccount2,
+                                    kNonexistentPermissionName),
+              checkStatelessInvalid)
+      .done();
+}
+
+/**
+ * C259 Grant permission, which exists in the system, but is not grantable
+ * @given an account with rights to grant rights to other accounts
+ * @when the account grants rights to an existing account
+ * AND a granted permission exists in the system, but is not grantable
+ * @then this transaction is stateless invalid
+ * AND it is not written in the block
+ */
+TEST_F(GrantPermissionTest, GrantNotGrantablePermission) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeAccountWithPerms(kAccount1,
+                                   kAccount1Keypair,
+                                   kCanGrantAll,
+                                   kRole1))
+      .skipProposal()
+      .skipBlock()
+      .sendTx(makeAccountWithPerms(kAccount2,
+                                   kAccount2Keypair,
+                                   kCanReceive,
+                                   kRole2))
+      .skipProposal()
+      .skipBlock()
+      .sendTx(accountGrantToAccount(kAccount1, kAccount1Keypair, kAccount2,
+                                    permissions::can_create_domain),
+              checkStatelessInvalid)
+      .done();
+}
+
+/**
+ * C260 Grant add signatory permission
+ * @given an account with rights to grant rights to other accounts
+ * AND the account grants add signatory rights to an existing account (permitee)
+ * @when the permitee adds signatory to the account
+ * @then a block with transaction to add signatory to the account is written
+ * AND there is a signatory added by the permitee
+ */
+TEST_F(GrantPermissionTest, GrantAddSignatoryPermission) {
```

Comment:
This and 4 following tests quite are hard to read thus understand
Can it be refactored somehow?